packages/components/src/unit-control/index.native.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { Text, View, TouchableWithoutFeedback } from 'react-native';
+/**
+ * Internal dependencies
+ */
+import RangeCell from '../mobile/bottom-sheet/range-cell';
+import StepperCell from '../mobile/bottom-sheet/stepper-cell';
+import Picker from '../mobile/picker';
+import styles from './style.scss';
+import { CSS_UNITS } from './utils';
+/**
+ * WordPress dependencies
+ */
+import { useRef } from '@wordpress/element';
+import { withPreferredColorScheme } from '@wordpress/compose';
+
+function UnitControl( {
+	currentInput,
+	label,
+	value,
+	onChange,
+	onUnitChange,
+	initialPosition,
+	min,
+	max,
+	separatorType,
+	units = CSS_UNITS,
+	unit,
+	getStylesFromColorScheme,
+	decimalNum,
+	...props
+} ) {
+	const pickerRef = useRef();
+
+	function onPickerSelect( selectedOption ) {
```

Comment:
Do you need that function? Why you don't use the `onUnitChange` directly?

Reply:
It looks like the leftover after switching back from `Menu` to  `BottomSheet`, will move it directly to the mentioned function.